src/main/java/com/mattermost/integration/figma/notification/service/FileNotificationService.java

```diff
@@ -1,62 +1,163 @@
     private static final String BASE_WEBHOOK_URL = "https://api.figma.com/v2/webhooks";
     private static final String PASSCODE = "Mattermost";
     private static final String FILE_COMMENT_EVENT_TYPE = "FILE_COMMENT";
     //Production Mattermost link
-    private static final String REDIRECT_URL = "http://localhost:8066/plugins/com.mattermost.apps/apps/spring-boot-example%s?secret=%s";
+    private static final String REDIRECT_URL = "http://localhost:8065/plugins/com.mattermost.apps/apps/spring-boot-example%s?secret=%s";
     private static final String FILE_COMMENT_URL = "/webhook/comment";
 
     private final RestTemplate restTemplate;
+    private final DMMessageService messageService;
+    private final KVService kvService;
+    private final JsonUtils jsonUtils;
+    private final FigmaWebhookService figmaWebhookService;
 
-    public FileNotificationService(RestTemplate restTemplate) {
+    public FileNotificationService(RestTemplate restTemplate, DMMessageService messageService, KVService kvService, JsonUtils jsonUtils, FigmaWebhookService figmaWebhookService) {
         this.restTemplate = restTemplate;
+        this.messageService = messageService;
+        this.kvService = kvService;
+        this.jsonUtils = jsonUtils;
+        this.figmaWebhookService = figmaWebhookService;
     }
 
-    public String subscribeToFileNotification(String teamId, String webhookSecret) {
+    public String subscribeToFileNotification(InputPayload inputPayload) {
+        String teamId = inputPayload.getValues().getTeamId();
         if (teamId != null && !teamId.isEmpty() && !teamId.isBlank()) {
-            HttpHeaders headers = new HttpHeaders();
-            headers.setContentType(MediaType.APPLICATION_JSON);
-            headers.set("Authorization", String.format("Bearer %s", FigmaTokenProvider.token.getAccessToken()));
-
-            FileCommentNotificationRequest fileCommentNotificationRequest = new FileCommentNotificationRequest();
-            fileCommentNotificationRequest.setEventType(FILE_COMMENT_EVENT_TYPE);
-            fileCommentNotificationRequest.setTeamId(teamId);
-            fileCommentNotificationRequest.setPasscode(PASSCODE);
-            //For production Mattermost link
-            //fileCommentNotificationRequest.setEndpoint(String.format(REDIRECT_URL, FILE_COMMENT_URL, webhookSecret));
-            fileCommentNotificationRequest.setEndpoint(NgrokLinkProvider.REDIRECT_URL.concat(FILE_COMMENT_URL));
-            log.debug("File notification request : " + fileCommentNotificationRequest);
-
-            HttpEntity<FileCommentNotificationRequest> request = new HttpEntity<>(fileCommentNotificationRequest, headers);
-            log.info("Sending comment request for team with id: " + teamId);
-            return restTemplate.postForEntity(BASE_WEBHOOK_URL, request, String.class).toString();
+            if (checkIfTeamWebhookIsPresent(teamId, inputPayload.getContext().getOauth2().getUser().getAccessToken())) {
```

Comment:
please consider to rename method like : hasWebhook according to java conventions